lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -435,73 +527,67 @@ def write(chunk)
             end
           end
 
-          ts_secs, ts_nanos = compute_timestamp(record, time)
+          ts_secs, ts_nanos = compute_timestamp(
+            entry_resource_type, record, time)
+          severity = compute_severity(
+            entry_resource_type, record, entry_common_labels)
+
           if @use_grpc
-            entry.metadata.timestamp = Google::Protobuf::Timestamp.new(
-              seconds: ts_secs,
-              nanos: ts_nanos
+            entry = Google::Logging::V2::LogEntry.new(
+              labels: entry_common_labels,
+              resource: Google::Api::MonitoredResource.new(
+                type: entry_resource_type,
+                labels: entry_resource_labels
+              ),
+              severity: grpc_severity(severity),
+              timestamp: Google::Protobuf::Timestamp.new(
+                seconds: ts_secs,
+                nanos: ts_nanos
+              )
             )
-
-            entry.metadata.severity =
-              grpc_severity(compute_severity(record, entry))
-
-            set_http_request_grpc(record, entry) # FIXME
+            set_http_request_grpc(record, entry)
+            set_payload_grpc(entry_resource_type, record, entry, is_json)
           else
-            entry.metadata.timestamp = {
-              seconds: ts_secs,
-              nanos: ts_nanos
-            }
-
-            entry.metadata.severity =
-              compute_severity(record, entry)
-
+            entry = Google::Apis::LoggingV2beta1::LogEntry.new(
+              labels: entry_common_labels,
```

Comment:
Moved this to the `LogEntry` level (The original implementation only set it there).